spring-cloud-gcp-pubsub-stream-binder/src/main/java/com/google/cloud/spring/stream/binder/pubsub/provisioning/PubSubChannelProvisioner.java

```diff
@@ -124,23 +128,45 @@ public void afterUnbindConsumer(ConsumerDestination destination) {
 		}
 	}
 
-	private Optional<Topic> ensureTopicExists(String topicName, boolean autoCreate) {
+	Topic ensureTopicExists(String topicName, boolean autoCreate) {
 		Topic topic = this.pubSubAdmin.getTopic(topicName);
-		if (topic == null) {
-			if (autoCreate) {
-				try {
-					topic = this.pubSubAdmin.createTopic(topicName);
-				}
-				catch (AlreadyExistsException alreadyExistsException) {
-					// Ignore concurrent topic creation - we're good as long as topic was created and exists
-					LOGGER.info("Failed to auto-create topic '" + topicName + "' because it already exists.");
-				}
+		if (topic != null) {
+			return topic;
+		}
+
+		if (autoCreate) {
+			try {
+				return this.pubSubAdmin.createTopic(topicName);
 			}
-			else {
-				throw new ProvisioningException("Non-existing '" + topicName + "' topic.");
+			catch (AlreadyExistsException alreadyExistsException) {
+				return ensureTopicExists(topicName, false);
```

Comment:
Maybe add a comment that this addresses a race condition between two microservices trying to autocreate?